main/src/ca/uwaterloo/flix/language/phase/Lowering.scala

```diff
@@ -796,7 +804,22 @@ object Lowering {
       val resultType = Types.Datalog
       LoweredAst.Expr.ApplyDef(defn.sym, argExps, List.empty, Types.MergeType, resultType, eff, loc)
 
-    case TypedAst.Expr.FixpointQueryWithProvenance(_, _, _, _, _, _) => ???
+    case TypedAst.Expr.FixpointQueryWithProvenance(exps, select, _, tpe, eff, loc) =>
```

Comment:
I think here around we should document the call we are trying to create:

```
provenanceOf(.., .., ...)
```

We can leave the lambda "abstract" and refer to its documentation